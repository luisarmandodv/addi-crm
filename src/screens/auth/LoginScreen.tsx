import React, { useState } from 'react';
import {
  Alert,
  View,
  Text,
  Image,
  TextInput,
  KeyboardAvoidingView,
  Platform,
  TouchableOpacity,
  ActivityIndicator,
  StyleSheet,
} from 'react-native';
import { FontAwesome } from '@expo/vector-icons';
import { LinearGradient } from 'expo-linear-gradient';
import { colors } from '@theme/colors';
import { formStyles, typography } from '@theme/globalStyles';
import Logo from '@assets/icon.png';
import { useLogin } from '@hooks/useLogin';

interface AuthInputProps {
  iconName: string;
  placeholder: string;
  value: string;
  onChangeText: (text: string) => void;
  secureTextEntry?: boolean;
}

interface LoginButtonProps {
  onPress: () => void;
  isLoading: boolean;
}

function Header() {
  return (
    <View style={styles.headerContainer}>
      <Image style={styles.headerImage} source={Logo} resizeMode="contain" />
      <Text style={styles.titleText}>Hello World</Text>
    </View>
  );
}

function AuthInput({ iconName, placeholder, value, onChangeText, secureTextEntry = false }: AuthInputProps) {
  return (
    <View style={formStyles.compactInputWrapper}>
      <FontAwesome name={iconName} size={24} color={colors.primaryBlack} style={styles.inputIcon} />
      <TextInput
        style={formStyles.compactInput}
        onChangeText={onChangeText}
        secureTextEntry={secureTextEntry}
        placeholder={placeholder}
        value={value}
      />
    </View>
  );
}

function LoginButton({ onPress, isLoading }: LoginButtonProps) {
  return (
    <TouchableOpacity style={styles.loginButton} onPress={onPress}>
      <Text style={styles.loginButtonText}>
        {isLoading ? <ActivityIndicator color={'#ffffff'} /> : 'Login'}
      </Text>
    </TouchableOpacity>
  );
}

export function LoginScreen() {
  const [email, onChangeEmail] = useState<string>('kminchelle');
  const [password, onChangePassword] = useState<string>('0lelplR');
  const { mutate, isLoading } = useLogin();

  function handleLogin() {
    if (!email || !password) {
      return Alert.alert('Rellena todos los campos');
    }
    mutate({ email, password });
  }

  return (
    <KeyboardAvoidingView
      behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
      style={{ flex: 1 }}
    >
      <LinearGradient
        colors={[colors.primary, colors.tertiary]}
        start={[0.0, 0.5]}
        end={[1.0, 0.5]}
        locations={[0.0, 1.0]}
        style={{ flex: 1 }}
      >
        <View style={styles.container}>
          <Header />
          <AuthInput iconName="user" placeholder="E-mail" value={email} onChangeText={onChangeEmail} />
          <AuthInput iconName="lock" placeholder="Password" value={password} onChangeText={onChangePassword} secureTextEntry={true} />
          <LoginButton onPress={handleLogin} isLoading={isLoading} />
        </View>
      </LinearGradient>
    </KeyboardAvoidingView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    marginVertical: 32,
    paddingHorizontal: 40,
  },
  headerContainer: {
    alignItems: 'center',
    marginBottom: 32,
  },
  headerImage: {
    width: 90,
    height: 60,
  },
  titleText: {
    ...typography.titleClean,
  },
  inputWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 16,
  },
  inputIcon: {
    marginRight: 16,
  },
  input: {
    flex: 1,
    fontSize: 16,
    backgroundColor: 'white'
  },
  loginButton: {
    alignItems: 'center',
    justifyContent: 'center',
    height: 48,
    backgroundColor: 'white',
    borderRadius: 24,
    marginTop: 40,
  },
  loginButtonText: {
    color: colors.primary,
    fontWeight: '700',
  },
});
