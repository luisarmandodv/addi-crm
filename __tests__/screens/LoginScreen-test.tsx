import React from 'react';
import { render, fireEvent, screen, waitFor } from '@testing-library/react-native';
import { LoginScreen } from '@screens/auth/LoginScreen';
import { QueryClient, QueryClientProvider } from "react-query";

jest.mock('@expo/vector-icons', () => ({
    FontAwesome: '',
    Feather: '',
}));

jest.mock('expo-linear-gradient', () => ({
    LinearGradient: '',
}));

describe('LoginScreen component', () => {
    const client = new QueryClient();

    it('Should render correctly component', () => {
        render(
            <QueryClientProvider client={client}>
                <LoginScreen />
            </QueryClientProvider>
        );
    });

    it('Should render snapshot', () => {
        render(
            <QueryClientProvider client={client}>
                <LoginScreen />
            </QueryClientProvider>
        );

        expect(screen.toJSON()).toMatchSnapshot();
    });
});
