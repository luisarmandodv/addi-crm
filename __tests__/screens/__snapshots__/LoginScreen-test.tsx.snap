// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen component Should render snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <
    colors={
      Array [
        "#82ccdd",
        "#60a3bc",
      ]
    }
    end={
      Array [
        1,
        0.5,
      ]
    }
    locations={
      Array [
        0,
        1,
      ]
    }
    start={
      Array [
        0,
        0.5,
      ]
    }
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "marginVertical": 32,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 40,
          }
        }
      >
        <View>
          <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "alignSelf": "center",
                "height": 60,
                "width": 90,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "marginBottom": 32,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 32,
                "textAlign": "center",
              }
            }
          >
            Hello World
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 16,
              "elevation": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "flex-start",
              "marginBottom": 8,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 1,
            }
          }
        >
          <
            color="#0c2461"
            name="user"
            size={24}
          />
          <TextInput
            keyboardType="email-address"
            onChangeText={[Function]}
            placeholder="E-mail"
            style={
              Object {
                "marginLeft": 12,
                "width": "100%",
              }
            }
            value=""
          />
        </View>
        <View
          style={
            Object {
              "marginBottom": 32,
              "marginTop": 8,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 16,
                "elevation": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "flex-start",
                "marginBottom": 8,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.15,
                "shadowRadius": 1,
              }
            }
          >
            <
              color="#0c2461"
              name="lock"
              size={24}
            />
            <TextInput
              maxLength={6}
              onChangeText={[Function]}
              placeholder="Password"
              secureTextEntry={true}
              style={
                Object {
                  "marginLeft": 12,
                  "width": "100%",
                }
              }
              value=""
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 40,
          "paddingHorizontal": 40,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "display": "flex",
            "height": 48,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#82ccdd",
              "fontWeight": "700",
            }
          }
        >
          Login
        </Text>
      </View>
    </View>
  </>
</View>
`;
